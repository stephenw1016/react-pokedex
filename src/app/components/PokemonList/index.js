// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { loadAllPokemon } from '../../actions';

import PokemonCard from '../PokemonCard';

type Props = {
  loadAllPokemon: Function,
  pokemon: Array<{id: number, name: string}>,
  requestPage: Function
};

type State = {
  pokemon: Array<{id: number, name: string}>
};

const mapStateToProps = ({pokemonList}) => {
  return {
    pokemon: pokemonList.pages[pokemonList.currentPage] || []
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    loadAllPokemon: dispatch(loadAllPokemon())
  };
};

class PokemonList extends React.Component<Props, State> {
  constructor (props: Props) {
    super(props);
  }

  render (): React.Node {
    return (
      <ul>
        {this.props.pokemon.map(pokemon =>
          <li key={pokemon.id}><PokemonCard pokemon={pokemon} /></li>)}
      </ul>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PokemonList);

