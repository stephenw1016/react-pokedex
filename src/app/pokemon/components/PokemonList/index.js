// @flow
import * as React from 'react';
import { connect } from 'react-redux';

import PokemonCard from '../PokemonCard';

type Props = {
  loadAllPokemon: Function,
  pokemon: Array<{id: number, name: string}>
};

export function PokemonList ({pokemon}: Props) {
  return (
    <ul>
      {pokemon.map(pokemonListItem =>
        <li key={pokemonListItem.id}>
          <PokemonCard pokemon={pokemonListItem} />
        </li>)}
    </ul>
  );
}

const mapStateToProps = ({pokemon: {pokemonList}}) => {
  return {
    pokemon: pokemonList.pages[pokemonList.currentPage] || []
  };
};

export default connect(mapStateToProps)(PokemonList);

