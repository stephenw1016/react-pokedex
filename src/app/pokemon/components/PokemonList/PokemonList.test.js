import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import {PokemonList} from '../PokemonList';

test('should render pokemon', () => {
  let pokemonList = [{ id: 1, name: 'bulbasaur' }, { id: 2, name: 'ivysaur' }, { id: 3, name: 'venusaur' }];
  let componentJSX = <MemoryRouter><PokemonList pokemon={pokemonList} /></MemoryRouter>;
  let componentJSON = renderer.create(componentJSX).toJSON();

  expect(componentJSON).toMatchSnapshot();
});
