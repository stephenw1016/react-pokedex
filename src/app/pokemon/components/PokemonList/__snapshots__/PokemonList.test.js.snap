// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render pokemon 1`] = `
<ul>
  <li>
    <div
      className="pokemon-card"
    >
      <div
        className="card-top"
      >
        <img
          src="http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/1.png"
        />
      </div>
      <div
        className="card-bottom"
      >
        <span>
          bulbasaur
        </span>
        <a
          href="/pokemon/1"
          onClick={[Function]}
        >
          Details
        </a>
      </div>
    </div>
  </li>
  <li>
    <div
      className="pokemon-card"
    >
      <div
        className="card-top"
      >
        <img
          src="http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/2.png"
        />
      </div>
      <div
        className="card-bottom"
      >
        <span>
          ivysaur
        </span>
        <a
          href="/pokemon/2"
          onClick={[Function]}
        >
          Details
        </a>
      </div>
    </div>
  </li>
  <li>
    <div
      className="pokemon-card"
    >
      <div
        className="card-top"
      >
        <img
          src="http://res.cloudinary.com/dwnebujkh/image/upload/v1473910425/pokemon/3.png"
        />
      </div>
      <div
        className="card-bottom"
      >
        <span>
          venusaur
        </span>
        <a
          href="/pokemon/3"
          onClick={[Function]}
        >
          Details
        </a>
      </div>
    </div>
  </li>
</ul>
`;
